nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/BulkStatus.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * (C) Copyright 2018 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Funsho David
+ */
+
+package org.nuxeo.ecm.core.bulk;
+
+import java.io.Serializable;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+/**
+ * This object holds the current bulk command execution status.
+ * <p/>
+ * This aggregates status and metrics of documentSet creation and operation computation.
+ *
+ * @since 10.2
+ */
+public class BulkStatus implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Possible states of bulk operation.
+     */
+    public enum State {
+        SCHEDULED, BUILDING, COMPLETED
+    }
+
+    protected UUID uuid;
+
+    protected BulkCommand command;
+
+    protected State state;
+
+    protected ZonedDateTime creationDate;
```

Comment:
ZonedDateTime vs Instant.  Any particular reason?

Reply:
Not really, I'm used to use ZonedDateTime when dealing with REST.

Reply:
Will change it to `Instant` in part 2

Reply:
+1 on `Instant`